proto/scheduler.proto

```diff
@@ -110,4 +112,44 @@ service Scheduler {
 service QueueExecutor {
   rpc EnqueueTaskReservation(EnqueueTaskReservationRequest)
       returns (EnqueueTaskReservationResponse) {}
+}
+
+message ExecutionNode {
```

Comment:
Nice! While you're here, mind adding a TODO to the `RegisterNodeRequest` above that we should eventually use this proto there? (Since it has all the same fields, we could eventually refactor it to look like this:

```proto
RegisterNodeRequest {
  ExecutionNode execution_node = 1;
}
```